pylearn2/models/mlp.py

```diff
@@ -22,11 +22,21 @@
 from theano.sandbox.cuda.dnn import dnn_available, dnn_pool
 from theano.tensor.signal.downsample import max_pool_2d
 import theano.tensor as T
+from theano.tensor.signal import downsample
+from theano.compat.six.moves import zip as izip
 
 from pylearn2.compat import OrderedDict
 from pylearn2.costs.mlp import Default
 from pylearn2.expr.probabilistic_max_pooling import max_pool_channels
-from pylearn2.linear import conv2d
+# Try to import the fast cudnn library, else fallback to conv2d
+cudnn2d = None
+if config.device == 'gpu':
```

Comment:
This will not work when users specify `'gpu0'` or `'gpu1'`, for instance.
A better test would be checking for `theano.sandbox.cuda.cuda_enabled` and `dnn_available`.